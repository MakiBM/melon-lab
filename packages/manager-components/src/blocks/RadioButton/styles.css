.radio-button {
  display: block;
  position: relative;
  padding-left: var(--space-l);
  cursor: pointer;
  line-height: 1;
  color: var(--color-primary-darkest);

  &:hover input:enabled ~ .radio-button__checkmark {
    background-color: var(--color-primary-darkest);
  }

  & input:checked ~ .radio-button__checkmark {
    background-color: var(--color-primary-darkest);
  }

  & input:checked ~ .radio-button__checkmark:after {
    display: block;
  }

  & input:disabled ~ .radio-button__text {
    text-decoration: line-through;
    color: var(--color-primary-darkest);
  }

  & input:disabled ~ .radio-button__checkmark {
    border-color: var(--color-primary-darkest);
    cursor: not-allowed;
  }

  & input:focus ~ .radio-button__checkmark {
    outline: auto;
  }
}

.radio-button__input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

.radio-button__checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 16px;
  width: 16px;
  border: 1px solid var(--color-primary-darkest);
  border-radius: 50%;
  transition-property: background-color;
  transition-duration: var(--transition-duration-default);

  &::after {
    content: '';
    position: absolute;
    display: none;
    top: 4px;
    left: 4px;
    width: 6px;
    height: 6px;
    border-radius: 50%;
    background: white;
  }
}
