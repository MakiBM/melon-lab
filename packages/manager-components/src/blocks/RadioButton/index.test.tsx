import React from 'react';
import RadioButton from './index';

describe('RadioButton', () => {
  const defaultElement = (
    <RadioButton
      name={'Default name'}
      value={'Default value'}
      text={'Default text'}
    />
  );

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be selectable by class "radio-button"', () => {
    expect(shallow(defaultElement).is('.radio-button')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(defaultElement).find('.radio-button').length).toBe(1);
  });

  it('should render to static HTML', () => {
    expect(render(defaultElement).text()).toEqual('Default text');
  });
});
