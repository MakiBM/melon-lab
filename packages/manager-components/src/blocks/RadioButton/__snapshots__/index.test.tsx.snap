// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should render correctly 1`] = `
<label
  className="jsx-__scopedHash radio-button"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <input
    className="jsx-__scopedHash radio-button__input"
    name="Default name"
    type="radio"
    value="Default value"
  />
  <span
    className="jsx-__scopedHash radio-button__checkmark"
  />
  <span
    className="jsx-__scopedHash radio-button__text"
  >
    Default text
  </span>
</label>
`;
